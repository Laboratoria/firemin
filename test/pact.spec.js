const pact = require('../lib/pact');


describe('pact', () => {

  it('should be a function', () => {
    expect(typeof pact).toBe('function');
  });

  it('should process tasks', (done) => {
    const tasks = [
      () => Promise.resolve(1),
      () => Promise.resolve(2),
      () => Promise.resolve(3),
      () => Promise.resolve(4),
      () => Promise.resolve(5),
    ];
    pact(tasks, 2).then((stats) => {
      expect(stats).toMatchSnapshot();
      done();
    });
  });

  it('should bail out on error in failFast mode', (done) => {
    const tasks = [
      () => Promise.resolve(1),
      () => Promise.resolve(2),
      () => Promise.reject('error!'),
      () => Promise.resolve(4),
      () => Promise.resolve(5),
    ];
    pact(tasks, 2).catch((err) => {
      expect(err).toBe('error!');
      done();
    });
  });

  it('should handle error when failFast is false', (done) => {
    const tasks = [
      () => Promise.resolve(1),
      () => Promise.resolve(2),
      () => Promise.reject('error!'),
      () => Promise.resolve(4),
      () => Promise.resolve(5),
    ];
    pact(tasks, 2, false).then((stats) => {
      expect(stats.results[2]).toBe('error!');
      expect(stats).toMatchSnapshot();
      done();
    });
  });

  describe('pact.splitArrayIntoBatches', () => {

    it('should be a function', () => {
      expect(typeof pact.splitArrayIntoBatches).toBe('function');
    });

    it('should split array into array of arrays, where each subarray has a given length limit', () => {
      expect(pact.splitArrayIntoBatches([1, 2, 3, 4, 5], 4)).toMatchSnapshot();
    });

  });

});
