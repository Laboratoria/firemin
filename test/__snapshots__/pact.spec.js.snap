// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pact pact.splitArrayIntoBatches should split array into array of arrays, where each subarray has a given length limit 1`] = `
Array [
  Array [
    1,
    2,
    3,
    4,
  ],
  Array [
    5,
  ],
]
`;

exports[`pact should handle error when failFast is false 1`] = `
Object {
  "batches": 3,
  "concurrency": 2,
  "results": Array [
    1,
    2,
    "error!",
    4,
    5,
  ],
  "total": 5,
}
`;

exports[`pact should process tasks 1`] = `
Object {
  "batches": 3,
  "concurrency": 2,
  "results": Array [
    1,
    2,
    3,
    4,
    5,
  ],
  "total": 5,
}
`;
